src/app/address/controllers/addressConfirm.js

```diff
@@ -3,17 +3,38 @@ const BaseController = require("hmpo-form-wizard").Controller;
 class AddressConfirmController extends BaseController {
   locals(req, res, callback) {
     super.locals(req, res, (err, locals) => {
-      const buildingNumber = req.sessionModel.get("addressLine1");
-      const streetName = req.sessionModel.get("addressLine2");
-      const town = req.sessionModel.get("addressTown");
-      const postcode = req.sessionModel.get("addressPostcode");
+      const sessionAddresses = req.sessionModel.get("addresses");
+      const addresses = JSON.parse(sessionAddresses);
 
-      const formattedAddress = `${buildingNumber}<br>${streetName},</br>${town},</br>${postcode},</br>`;
+      const currentAddress = addresses.shift();
+      const formattedCurrentAddress = this.formatAddress(currentAddress);
+      locals.formattedAddress = formattedCurrentAddress;
+
+      // format previous addresses
+      if (addresses.length) {
+        const formattedPreviousAddresses = addresses.map((address) => {
+          return this.formatAddress(address);
+        });
+
+        locals.previousAddresses = formattedPreviousAddresses;
+      }
 
-      locals.formattedAddress = formattedAddress;
       callback(null, locals);
     });
   }
+
+  saveValues(req, res, callback) {
+    super.saveValues(req, res, () => {
+      // temporary until add previous addresses has been confirmed.
+      const addPreviousAddresses = req.body.addPreviousAddresses !== null;
+      req.sessionModel.set("addPreviousAddresses", addPreviousAddresses);
```

Comment:
This is an not great looking button on the confirmation page that WILL be subject to change once the previous address journey has been finalised.